### PR TITLE
add fno-offload-uniform-block flag for rocm5.7 and up

### PR DESCRIPTION
**Pull request by illsilin**
This should take care of the changes in the default compiler settings starting from rocm version 5.7.

### CODE REVIEW

**Comment by junliume** [comment]
@illsilin this PR is to fix https://github.com/ROCmSoftwarePlatform/composable_kernel/issues/837

Meanwhile, could you add comment in the code such as:
```
# No assumption that HIP kernels are launched with uniform block size for backward compatibility
# SWDEV-413293 and https://reviews.llvm.org/D155213
```

**Comment by illsilin** [comment]
OK, sure.

**Review by junliume — APPROVED** [review]

**Review by junliume — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -89,6 +89,14 @@ else()
     message("Building CK for the following targets: ${AMDGPU_TARGETS}")
 endif()
 find_package(hip)
+# No assumption that HIP kernels are launched with uniform block size for backward compatibility
+# SWDEV-413293 and https://reviews.llvm.org/D155213
+math(EXPR hip_VERSION_FLAT "(${hip_VERSION_MAJOR} * 1000 + ${hip_VERSION_MINOR}) * 100000 + ${hip_VERSION_PATCH}")
+message("hip_version_flat=${hip_VERSION_FLAT}")
+if(${hip_VERSION_FLAT} GREATER 500700000)
```
Comment (junliume) [inline CMakeLists.txt:96]: @illsilin effective HIP_FLAT_VERSION 500723302, so preferably this line should be 
```
if(${hip_VERSION_FLAT} GREATER_EQUAL 500723302)
```
This defect will impact a few internal dockers with HIP version between 500700000 and 500723302

Comment (illsilin) [reply CMakeLists.txt:96]: Will update the threshold version.

**Comment by atamazov** [comment]
@junliume @illsilin I do not recommend adding the new option unconditionally, see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2307#issuecomment-1671699031

**Comment by illsilin** [comment]
I can run the full QA suite with rocm5.7 RC1 and this flag and see if there is any performance degradation.

**Comment by atamazov** [comment]
@illsilin That would be very good, thanks! Please update us with the results.

**Comment by illsilin** [comment]
Apparently, the rocm5.7 RC1 is the threshold version so the flag does not get set for it:
hip_version_flat=500723302
We can wait for the RC2 to do this, or I can try to run the tests on a development node with the latest rocm build.

**Comment by atamazov** [comment]
@illsilin FYI this is how we use check_cxx_compiler_flag() in MIOpen (more future-proof solution): https://github.com/ROCm/MIOpen/pull/2741

**Pull request merged**